api/processor_api.py

```diff
@@ -0,0 +1,118 @@
+import time
+from threading import Thread
+from queue import Queue
+from multiprocessing.managers import BaseManager
+from fastapi import FastAPI
+from fastapi.staticfiles import StaticFiles
+from fastapi.responses import FileResponse, StreamingResponse
+import uvicorn
+import os
+import logging
+from typing import Dict
+from pydantic import BaseModel
+from typing import Optional
+from api.config_keys import Config,APP,DETECTOR,POSTPROCESSOR,LOGGER,API_
+from share.commands import Commands
+
+logger = logging.getLogger(__name__)
+
+class QueueManager(BaseManager): pass
+
+class ProcessorAPI:
+    """
+    The Webgui object implements a fastapi application and acts as an interface for users.
+    Once it is created it will act as a central application for viewing outputs.
+
+    :param config: Is a ConfigEngine instance which provides necessary parameters.
+    :param engine_instance:  A ConfigEngine object which store all of the config parameters. Access to any parameter
+        is possible by calling get_section_dict method.
+    """
+
+    def __init__(self, config):
+        self.config = config
+        self._setup_queues()
+        self._host = self.config.get_section_dict("API")["Host"]
+        self._port = int(self.config.get_section_dict("API")["Port"])
+        self.app = self.create_fastapi_app()
+
+    def _setup_queues(self):
+        QueueManager.register('get_cmd_queue')
+        QueueManager.register('get_result_queue')
+        self._queue_host = self.config.get_section_dict("CORE")["Host"]
+        self._queue_port = int(self.config.get_section_dict("CORE")["QueuePort"])
+        auth_key = self.config.get_section_dict("CORE")["QueueAuthKey"]
+        self._queue_manager = QueueManager(address=(self._queue_host, self._queue_port), authkey=auth_key.encode('ascii'))
+        
+        while True:
+            try:
+                self._queue_manager.connect()
+                break
+            except ConnectionRefusedError:
+                logger.warning("Waiting for core's queue to initiate ... ")
+                time.sleep(1)
+
+        logger.info("Connection established to Core's queue")
+        self._cmd_queue = self._queue_manager.get_cmd_queue()
+        self._result_queue = self._queue_manager.get_result_queue()
+
+    def create_fastapi_app(self):
+        # Create and return a fastapi instance
+        app = FastAPI()
+
+        if os.environ.get('DEV_ALLOW_ALL_ORIGINS', False):
```

Comment:
@mats-claassen @mhsekhavat @emma-w-dev is this still relevant here? Assuming that the react GUI has been moved out of the processor alltogether.